drivers/include/onewire.h

```diff
@@ -0,0 +1,246 @@
+ * platform does not support this GPIO mode.
+ *
+ * @param[out] owi      1-Wire bus device descriptor
+ * @param[in] params    configuration parameters
+ *
+ * @return  ONEWIRE_OK on success
+ * @return  ONEWIRE_ERR_PINCFG if open-drain mode unsupported
+ */
+int onewire_init(onewire_t *owi, const onewire_params_t *params);
+
+/**
+ * @brief   Generate a bus reset sequence and look for connected devices
+ *
+ * Every bus transaction must be started with this reset sequence. During this
+ * sequence it is detected, if there are any slaves connected to the bus.
+ *
+ * @param[in] owi       1-Wire bus device descriptor
+ * @param[in] rom       1-Wire ROM code of target device
+ *
+ * @return  ONEWIRE_OK if slave(s) were found
+ * @return  ONEWIRE_NODEV if no slave answered the reset sequence
+ */
+int onewire_reset(const onewire_t *owi, const onewire_rom_t *rom);
+
+/**
+ * @brief   Read data from the bus
+ *
+ * @param[in] owi       1-Wire bus device descriptor
+ * @param[out] data     buffer to write received bytes into
+ * @param[in] len       number of bytes to read from the bus
+ */
+void onewire_read(const onewire_t *owi, void *data, size_t len);
+
+/**
+ * @brief   Write data to the bus
+ *
+ * @param[in] owi       1-Wire bus device descriptor
+ * @param[in] data      buffer holding the data that is written to the bus
+ * @param[in] len       number of bytes to write to the bus
+ */
+void onewire_write(const onewire_t *owi, const void *data, size_t len);
+
+/**
+ * @brief   Convenience function for writing a single byte to the bus
+ *
+ * @param[in] owi       1-Wire bus device descriptor
+ * @param[in] data      data byte to write to the bus
+ */
+static inline void onewire_write_byte(const onewire_t *owi, uint8_t data)
+{
+    onewire_write(owi, &data, 1);
+}
+
+/**
+ * @brief   Calculate a 8-bit CRC using the 1-Wire specific polynomial
+ *
+ * @param[in] data      input data
+ * @param[in] len       len of @p data in bytes
+ *
+ * @return  calculate 8-bit CRC
+ */
+uint8_t onewire_crc8(const uint8_t *data, size_t len);
+
+/**
+ * @brief   Search for devices on the given 1-Wire bus
+ *
+ * @see     https://pdfserv.maximintegrated.com/en/an/AN937.pdf page 51+52
+ *
+ * Use this function to discover devices connected to the given 1-Wire bus. To
+ * carry out a full device discovery, this function has to be called multiple
+ * times. For each iteration, the return value of the previous call as well as
+ * the previous discovered ROM code have to be given as @p rom and @p ld
+ * parameters to the subsequent call. One the function returns 0, the last
+ * device has been found and there are no further devices to be discovered.
+ *
+ * @param[in] owi       1-Wire bus
+ * @param[in,out] rom   next discovery ROM code, needs to hold previous value
+ *                      when function is called multiple times
+ * @param[in] ld        position of last discrepancy or ONEWIRE_SEARCH_FIRST for
```

Comment:
discrepancy?

Reply:
wording used in the description of the discovery process as used in the pdf linked above.